tmt/steps/prepare/artifact/providers/koji.py

```diff
@@ -184,3 +234,136 @@ def _download_artifact(
             )
         except Exception as error:
             raise DownloadError(f"Failed to download '{artifact}'.") from error
+
+    def make_rpm_artifact(self, rpm_meta: dict[str, str]) -> RpmArtifactInfo:
+        """
+        Create a normal build RPM artifact from metadata returned by listBuildRPMs.
+        """
+        name = rpm_meta["name"]
+        version = rpm_meta["version"]
+        release = rpm_meta["release"]
+        arch = rpm_meta["arch"]
+
+        # Construct the full URL for this RPM
+        url = (
+            f"{self._topurl}/packages/{name}/"
+            f"{version}/{release}/{arch}/"
+            f"{name}-{version}-{release}.{arch}.rpm"
+        )
+
+        return RpmArtifactInfo(_raw_artifact={**rpm_meta, "url": url})
+
+    def make_scratch_artifact(self, task_id: int, filename: str) -> KojiScratchRpmArtifactInfo:
+        """
+        Create a scratch RPM artifact from a task output filename.
+        """
+        pathinfo = koji.PathInfo(  # type: ignore[union-attr]
+            topdir=self._topurl
+        )
+        work_path = pathinfo.work("DEFAULT")
+        task_path = pathinfo.taskrelpath(task_id)
+        url = f"{work_path}/{task_path}/{filename}"
+
+        raw_artifact = {
+            "filename": filename,
+            "url": url,
+        }
+        return KojiScratchRpmArtifactInfo(_raw_artifact=raw_artifact)
+
+
+@provides_artifact_provider("koji.task")  # type: ignore[arg-type]
+class KojiTask(KojiArtifactProvider):
```

Comment:
I am split between if the class name should be needlessly redundant like `KojiTaskArtifactProvider`, it is the default naming convention so far, but I can see how dropping these can be cleaner when you reference by namespace. `PLC0415` is slightly relevant to this and should open an issue for discussion later.

As for here, let's see if someone else also complains, otherwise, I wouldn't mind as-is.

Reply:
I would complain, but I would like to merge it as-is, and fix the naming later, with the repository file provider in as well. It would be easier to establish some naming scheme and fix them all in one go.